src/index.html

```diff
@@ -7,6 +7,40 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar-wrapper">
+      <main class="calendar calendar--month-length--29">
+        <div class="calendar__day calendar__day--1 calendar__day--start-day-sunday"></div>
```

Comment:
The calendar__day--start-day-$$$ modifier should be applied to the calendar container, not to the first element. 
In your case it should be applied to the "main" tag.